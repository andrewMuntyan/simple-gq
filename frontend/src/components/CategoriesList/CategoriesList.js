import React from 'react';
import PropTypes from 'prop-types';

import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';

import { ListRow } from '../ListRow';
import { List } from '../List';

export const GET_CATEGORIES = gql`
  query GET_CATEGORIES {
    categories {
      id
      name
    }
  }
`;

const CategoriesList = ({ onCategoryPick, selectedCategory }) => {
  const { data, loading, error, fetchMore } = useQuery(GET_CATEGORIES);
  const { categories = [] } = data;

  if (loading) {
    return <h2>Loading Categories...</h2>;
  }

  // TODO: add spinner
  return categories.length ? (
    <List
      itemsData={categories}
      RowRenderer={ListRow}
      onItemClick={onCategoryPick}
      selectedItem={selectedCategory}
    />
  ) : (
    <h2>There is Nothing... Add new Category!</h2>
  );
};

CategoriesList.propTypes = {
  onCategoryPick: PropTypes.func.isRequired,
  selectedCategory: PropTypes.string
};
CategoriesList.defaultProps = {
  selectedCategory: null
};

export default CategoriesList;
