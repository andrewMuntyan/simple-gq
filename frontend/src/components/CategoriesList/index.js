export { default as CategoriesList } from './CategoriesList';
export { GET_CATEGORIES } from './CategoriesList';
