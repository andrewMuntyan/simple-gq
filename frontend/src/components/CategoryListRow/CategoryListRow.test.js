import React from 'react';

import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { CategoryListRow, DELETE_CATEGORY } from '.';
import { GET_CATEGORIES } from '../CategoriesList';

import {
  fakeCategory,
  waitAndUpdateWrapper,
  fakeGQError,
  getMockedStaff
} from '../../testUtils';

const categoryData = fakeCategory();
const { name: removedCategoryName } = categoryData;
const mutationVariables = {
  ok: { name: removedCategoryName },
  fail: { name: 'error' }
};
const removedCategoryResponseData = {
  deleteCategory: categoryData
};

const getMocks = ({ isError = false } = {}) => [
  {
    request: {
      query: DELETE_CATEGORY,
      variables: mutationVariables.ok
    },
    result: jest.fn(() =>
      isError ? fakeGQError : { data: removedCategoryResponseData }
    )
  }
];

const cacheInitialQueries = [
  {
    query: GET_CATEGORIES,
    data: { categories: [categoryData] }
  }
];

// TODO: Do not want to use MUI testing tools like createRender() or smth
// but have to make selection better. Add unique attr or smth.
const deleteBtnSelector = 'button';

describe('<CategoryListRow />', () => {
  it('renders and maches snapshot', () => {
    const { MockedAppProvider } = getMockedStaff();
    const wrapper = mount(
      <MockedAppProvider>
        <CategoryListRow data={categoryData} />
      </MockedAppProvider>
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should set selectedCategory to AppState', () => {
    const { MockedAppProvider, setAppStateMock } = getMockedStaff();
    const wrapper = mount(
      <MockedAppProvider>
        <CategoryListRow data={categoryData} />
      </MockedAppProvider>
    );

    expect(setAppStateMock).toHaveBeenCalledTimes(0);

    wrapper.simulate('click');

    expect(setAppStateMock).toHaveBeenCalledTimes(1);
    expect(setAppStateMock).toHaveBeenCalledWith({
      selectedCategory: categoryData.name,
      searchTerm: ''
    });
  });

  it('calls the mutation and updates local cache', async () => {
    const mocks = getMocks();
    const { onActionDone, cache, MockedAppProvider } = getMockedStaff({
      mocks,
      cacheInitialQueries
    });

    // cache initialised with array
    const { categories: initialCategoriesData } = cache.readQuery({
      query: GET_CATEGORIES
    });
    expect(initialCategoriesData).toBeInstanceOf(Array);
    expect(initialCategoriesData).toHaveLength(1);
    expect(initialCategoriesData[0]).toEqual(categoryData);

    const wrapper = mount(
      <MockedAppProvider>
        <CategoryListRow data={categoryData} />
      </MockedAppProvider>
    );

    // delete button is rendered
    const deleteBtn = wrapper.find(deleteBtnSelector);
    expect(deleteBtn).toHaveLength(1);

    // deleteBtn clicked
    deleteBtn.simulate('click');

    await waitAndUpdateWrapper(wrapper);

    // Check if mutation has been called
    expect(mocks[0].result).toHaveBeenCalledTimes(1);

    // Check if success message was shown
    expect(onActionDone).toHaveBeenCalledTimes(1);
    expect(onActionDone).toHaveBeenCalledWith({
      data: removedCategoryResponseData
    });

    await waitAndUpdateWrapper(wrapper);

    // cache is updated with data from mocked response
    const { categories: resultedCategoriesData } = cache.readQuery({
      query: GET_CATEGORIES
    });

    expect(resultedCategoriesData).toBeInstanceOf(Array);
    expect(resultedCategoriesData).toHaveLength(0);
  });

  it('correctly reacts on an graphql error', async () => {
    const mocks = getMocks({ isError: true });
    const { onActionDone, MockedAppProvider } = getMockedStaff({
      mocks,
      cacheInitialQueries
    });
    const wrapper = mount(
      <MockedAppProvider>
        <CategoryListRow data={categoryData} />
      </MockedAppProvider>
    );

    // delete button is rendered
    const deleteBtn = wrapper.find(deleteBtnSelector);
    // deleteBtn clicked
    deleteBtn.simulate('click');

    await waitAndUpdateWrapper(wrapper);

    // Check if mutation has been called
    expect(mocks[0].result).toHaveBeenCalledTimes(1);

    // Check if error message was shown
    expect(onActionDone).toHaveBeenCalledTimes(1);
    const error = onActionDone.mock.calls[0][0];
    expect(error.toString()).toEqual(
      'Error: GraphQL error: invalid input data'
    );
  });
});
