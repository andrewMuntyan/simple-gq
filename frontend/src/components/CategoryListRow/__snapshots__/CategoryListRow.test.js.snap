// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryListRow /> renders and maches snapshot 1`] = `
<MockedAppProvider>
  <MockedProvider
    addTypename={true}
    cache={
      InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      }
    }
    mocks={Array []}
  >
    <ApolloProvider
      client={
        ApolloClient {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "clearStoreCallbacks": Array [],
          "defaultOptions": Object {},
          "disableNetworkFetches": false,
          "link": MockLink {
            "addTypename": true,
            "mockedResponsesByKey": Object {},
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutate": [Function],
          "query": [Function],
          "queryDeduplication": true,
          "queryManager": QueryManager {
            "assumeImmutableResults": false,
            "clientAwareness": Object {
              "name": undefined,
              "version": undefined,
            },
            "dataStore": DataStore {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
            },
            "fetchQueryRejectFns": Map {},
            "idCounter": 1,
            "inFlightLinkObservables": Map {},
            "link": MockLink {
              "addTypename": true,
              "mockedResponsesByKey": Object {},
            },
            "localState": LocalState {
              "cache": InMemoryCache {
                "addTypename": true,
                "cacheKeyRoot": KeyTrie {
                  "weakness": true,
                },
                "config": Object {
                  "addTypename": true,
                  "dataIdFromObject": [Function],
                  "fragmentMatcher": HeuristicFragmentMatcher {},
                  "freezeResults": false,
                  "resultCaching": true,
                },
                "data": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "maybeBroadcastWatch": [Function],
                "optimisticData": DepTrackingCache {
                  "data": Object {},
                  "depend": [Function],
                },
                "silenceBroadcast": false,
                "storeReader": StoreReader {
                  "executeSelectionSet": [Function],
                  "executeStoreQuery": [Function],
                  "executeSubSelectedArray": [Function],
                  "freezeResults": false,
                },
                "storeWriter": StoreWriter {},
                "typenameDocumentCache": Map {},
                "watches": Set {},
              },
              "client": [Circular],
            },
            "mutationStore": MutationStore {
              "store": Object {},
            },
            "onBroadcast": [Function],
            "pollingInfoByQueryId": Map {},
            "queries": Map {},
            "queryDeduplication": true,
            "queryStore": QueryStore {
              "store": Object {},
            },
            "ssrMode": false,
            "transformCache": WeakMap {},
          },
          "reFetchObservableQueries": [Function],
          "resetStore": [Function],
          "resetStoreCallbacks": Array [],
          "store": DataStore {
            "cache": InMemoryCache {
              "addTypename": true,
              "cacheKeyRoot": KeyTrie {
                "weakness": true,
              },
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {},
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {},
            },
          },
          "typeDefs": undefined,
          "version": "2.6.4",
          "watchQuery": [Function],
        }
      }
    >
      <ListRow
        data={
          Object {
            "__typename": "Category",
            "createdAt": "2019-09-28T11:58:36.949Z",
            "id": "uuid1",
            "name": "category 1",
            "updatedAt": "2019-09-28T11:58:38.949Z",
          }
        }
      >
        <WithStyles(ForwardRef(ListItem))
          button={true}
          className="makeStyles-item-1"
          onClick={[Function]}
          selected={false}
        >
          <ForwardRef(ListItem)
            button={true}
            className="makeStyles-item-1"
            classes={
              Object {
                "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                "button": "MuiListItem-button",
                "container": "MuiListItem-container",
                "dense": "MuiListItem-dense",
                "disabled": "Mui-disabled",
                "divider": "MuiListItem-divider",
                "focusVisible": "Mui-focusVisible",
                "gutters": "MuiListItem-gutters",
                "root": "MuiListItem-root",
                "secondaryAction": "MuiListItem-secondaryAction",
                "selected": "Mui-selected",
              }
            }
            onClick={[Function]}
            selected={false}
          >
            <WithStyles(ForwardRef(ButtonBase))
              className="MuiListItem-root makeStyles-item-1 MuiListItem-gutters MuiListItem-button"
              component="div"
              disabled={false}
              focusVisibleClassName="Mui-focusVisible"
              onClick={[Function]}
            >
              <ForwardRef(ButtonBase)
                className="MuiListItem-root makeStyles-item-1 MuiListItem-gutters MuiListItem-button"
                classes={
                  Object {
                    "disabled": "Mui-disabled",
                    "focusVisible": "Mui-focusVisible",
                    "root": "MuiButtonBase-root",
                  }
                }
                component="div"
                disabled={false}
                focusVisibleClassName="Mui-focusVisible"
                onClick={[Function]}
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root makeStyles-item-1 MuiListItem-gutters MuiListItem-button"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <WithStyles(ForwardRef(ListItemText))
                    key=".0"
                    primary="category 1"
                  >
                    <ForwardRef(ListItemText)
                      classes={
                        Object {
                          "dense": "MuiListItemText-dense",
                          "inset": "MuiListItemText-inset",
                          "multiline": "MuiListItemText-multiline",
                          "primary": "MuiListItemText-primary",
                          "root": "MuiListItemText-root",
                          "secondary": "MuiListItemText-secondary",
                        }
                      }
                      primary="category 1"
                    >
                      <div
                        className="MuiListItemText-root"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="MuiListItemText-primary"
                          component="span"
                          variant="body1"
                        >
                          <ForwardRef(Typography)
                            className="MuiListItemText-primary"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            component="span"
                            variant="body1"
                          >
                            <span
                              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                            >
                              category 1
                            </span>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                      </div>
                    </ForwardRef(ListItemText)>
                  </WithStyles(ForwardRef(ListItemText))>
                  <WithStyles(ForwardRef(IconButton))
                    aria-label="delete"
                    color="secondary"
                    key=".1"
                    onClick={[Function]}
                    size="small"
                  >
                    <ForwardRef(IconButton)
                      aria-label="delete"
                      classes={
                        Object {
                          "colorInherit": "MuiIconButton-colorInherit",
                          "colorPrimary": "MuiIconButton-colorPrimary",
                          "colorSecondary": "MuiIconButton-colorSecondary",
                          "disabled": "Mui-disabled",
                          "edgeEnd": "MuiIconButton-edgeEnd",
                          "edgeStart": "MuiIconButton-edgeStart",
                          "label": "MuiIconButton-label",
                          "root": "MuiIconButton-root",
                          "sizeSmall": "MuiIconButton-sizeSmall",
                        }
                      }
                      color="secondary"
                      onClick={[Function]}
                      size="small"
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        aria-label="delete"
                        centerRipple={true}
                        className="MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                        disabled={false}
                        focusRipple={true}
                        onClick={[Function]}
                      >
                        <ForwardRef(ButtonBase)
                          aria-label="delete"
                          centerRipple={true}
                          className="MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          disabled={false}
                          focusRipple={true}
                          onClick={[Function]}
                        >
                          <button
                            aria-label="delete"
                            className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeSmall"
                            disabled={false}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={0}
                            type="button"
                          >
                            <span
                              className="MuiIconButton-label"
                            >
                              <ForwardRef>
                                <WithStyles(ForwardRef(SvgIcon))>
                                  <ForwardRef(SvgIcon)
                                    classes={
                                      Object {
                                        "colorAction": "MuiSvgIcon-colorAction",
                                        "colorDisabled": "MuiSvgIcon-colorDisabled",
                                        "colorError": "MuiSvgIcon-colorError",
                                        "colorPrimary": "MuiSvgIcon-colorPrimary",
                                        "colorSecondary": "MuiSvgIcon-colorSecondary",
                                        "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                        "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                        "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                        "root": "MuiSvgIcon-root",
                                      }
                                    }
                                  >
                                    <svg
                                      aria-hidden="true"
                                      className="MuiSvgIcon-root"
                                      focusable="false"
                                      role="presentation"
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                                      />
                                    </svg>
                                  </ForwardRef(SvgIcon)>
                                </WithStyles(ForwardRef(SvgIcon))>
                              </ForwardRef>
                            </span>
                            <NoSsr>
                              <WithStyles(undefined)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(undefined)>
                            </NoSsr>
                          </button>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(IconButton)>
                  </WithStyles(ForwardRef(IconButton))>
                  <NoSsr>
                    <WithStyles(undefined)
                      center={false}
                    >
                      <ForwardRef(TouchRipple)
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(undefined)>
                  </NoSsr>
                </div>
              </ForwardRef(ButtonBase)>
            </WithStyles(ForwardRef(ButtonBase))>
          </ForwardRef(ListItem)>
        </WithStyles(ForwardRef(ListItem))>
      </ListRow>
    </ApolloProvider>
  </MockedProvider>
</MockedAppProvider>
`;
